src/robot/libraries/Remote.py

```diff
@@ -227,23 +228,49 @@ def _server(self):
         finally:
             server('close')()
 
+    @property
+    def _kw_cache(self):
+        """
+        returns False or a dict with structure:
+        { "keyword_name" : { 'args':[], 'tags':[], 'doc':"", 'types':[]},
+          "kw2"          : {...}, etc. }
+        """
+        if self.__kw_cache is None:
+            self.__kw_cache = False
+            try:
+                with self._server as server:
+                    self.__kw_cache = server.get_library_information()
+            except:
```

Comment:
Too broad exception, what are usual expected exceptions here? (in case of not supported version etc)

Reply:
I figured there were 2 possible ways to go here.
1. Check if get_library_information is implemented. If so, use it and let it fail whenever it does.
2. Just try to call it and fall back to the _old way_ if it doesn't work, for whatever reason.

I went for the latter , because I thought it stayed closest to the existing behaviour. Not a fan of broad exception handling myself, so we could still opt for the first option.

Reply:
Introspection isn't available, so checking if it is implemented can't be done. But as it turns out the exceptions were already narrowed down on another level, that also contained a retry. Now the try on get_library_info replaces the first attempt and only the specific error is caught.